lib/pacemaker/pcmk_sched_primitive.c

```diff
@@ -1636,7 +1636,7 @@ ban_if_not_locked(gpointer data, gpointer user_data)
     const pcmk_node_t *node = (const pcmk_node_t *) data;
     pcmk_resource_t *rsc = (pcmk_resource_t *) user_data;
 
-    if (strcmp(node->details->uname, rsc->lock_node->details->uname) != 0) {
+    if (!pcmk__same_node(node, rsc->private->lock_node)) {
```

Comment:
probably better but not equivalent

Reply:
well there is the possibility of two nodes with the same name (more of a risk in old heartbeat or Corosync 1 clusters). but that would break many things in the scheduler. at some point it would be worthwhile to handle that case more explicitly throughout the scheduler

Reply:
Yeah that's what I was getting at, and that makes sense